server/py/services/alerts/crud/alerts.py

```diff
@@ -410,24 +410,23 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
         if alert.reset_policy == mlrun.common.schemas.alert.ResetPolicy.MANUAL:
             # before resetting an alert, update number_of_events if required
             number_of_events = self._get_number_of_events(alert.id)
-            # if they are equal, the number_of_events is already set to the correct value
-            if number_of_events > alert.criteria.count:
-                alert_state = (
-                    framework.utils.singletons.db.get_db().get_alert_state_dict(
-                        session, alert.id
-                    )
-                )
-                activation_time = alert_state.get("last_updated")
-                activation_id = alert_state.get("full_object", {}).get(
-                    "last_activation_id"
+            alert_state = framework.utils.singletons.db.get_db().get_alert_state_dict(
+                session, alert.id
+            )
+            activation_time = alert_state.get("last_updated")
+            activation_id = alert_state.get("full_object", {}).get("last_activation_id")
+            if activation_time and activation_id:
```

Comment:
If one or both are None - is it a valid case?
If not, we should at least log a warning.

Reply:
@TomerShor nope - not a valid case, will add warning